docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+    please cite as described in 'parallel --citation'.
+
+    (.dev) ~ yq --version
+    yq (https://github.com/mikefarah/yq/) version v4.44.3
+    ```
+
+### Scaling Strategies
+
+#### Device-Based Parallelization
+
+In this approach, we run ANTA instances in parallel, with each instance handling a specific device. This strategy is effective when:
+
+- Each device has unique test requirements, i.e., one catalog per device
+- You want to maintain separate results per device
+- You need fine-grained control over test execution
+
+Example inventory structure:
+
+```yaml
+anta_inventory:
+  hosts:
+  - host: 172.20.20.101
+    name: DC1-SPINE1
+  - host: 172.20.20.102
+    name: DC1-SPINE2
+  # ... more devices
+```
+
+Example test catalog structure (one catalog per device):
+
+```yaml
+# DC1-SPINE2.yml
+anta.tests.connectivity:
+- VerifyLLDPNeighbors:
+    neighbors:
+    - neighbor_device: DC1-LEAF1A
+      neighbor_port: Ethernet1
+      port: Ethernet1
+    result_overwrite:
+      custom_field: 'Local: Ethernet1 - Remote: DC1-LEAF1A Ethernet1'
+
+# DC1-SPINE1.yml
+anta.tests.connectivity:
+- VerifyLLDPNeighbors:
+    neighbors:
+    - neighbor_device: DC1-LEAF1A
+      neighbor_port: Ethernet2
+      port: Ethernet1
+    result_overwrite:
+      custom_field: 'Local: Ethernet1 - Remote: DC1-LEAF1A Ethernet2'
+```
+
+#### Tag-Based Parallelization
+
+This approach runs parallel instances based on device tags, useful when:
+
+- Devices in the same role share similar test requirements
+- You want to test entire PODs or device groups simultaneously
+- You need to organize tests by network function or location
+
+Example inventory structure:
+
+```yaml
+---
+anta_inventory:
+  hosts:
+  - host: 172.20.20.101
+    name: DC1-SPINE1
+    tags: ["SPINE", "DC1"]
+
+  - host: 172.20.20.102
+    name: DC1-SPINE2
+    tags: ["SPINE", "DC1"]
+
+  - host: 172.20.20.111
+    name: DC1-LEAF1A
+    tags: ["LEAF", "DC1"]
+
+  - host: 172.20.20.112
+    name: DC1-LEAF1B
+    tags: ["LEAF", "DC1"]
+
+  - host: 172.20.20.201
+    name: DC2-SPINE1
+    tags: ["SPINE", "DC2"]
+
+  - host: 172.20.20.202
+    name: DC2-SPINE2
+    tags: ["SPINE", "DC2"]
+  # ... more devices
+```
+
+Example test catalog structure (one catalog for all devices):
+
+```yaml
+anta.tests.vxlan:
+  - VerifyVxlan1Interface:
+      filters:
+        tags: ["BL", "LEAF"]
+  - VerifyVxlanConfigSanity:
+      filters:
+        tags: ["BL", "LEAF"]
+
+anta.tests.routing:
+  generic:
+    - VerifyRoutingProtocolModel:
+        model: multi-agent
+        filters:
+          tags: ["SPINE"]
+```
+
+#### Optimizing Catalog Distribution
+
+For large fabrics, consider:
+
+1. Creating separate catalogs per device type
+2. Organizing catalogs by POD or datacenter
+3. Using tag-based filtering within catalogs
+
+Benefits:
+
+- Reduced memory usage per ANTA instance
+- Faster catalog processing
+- More efficient test execution
+
+### Implementation
+
+#### Device-Based Implementation
```

Comment:
I would move Implementation under each stratgey in a subtitle for easier navigation